core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/HadoopConfigurationService.java

```diff
@@ -19,37 +25,109 @@
+    public static final String YARN_NM_LOCAL_DIRS = "yarn.nodemanager.local-dirs";
+    public static final String YARN_NM_LOG_DIRS = "yarn.nodemanager.log-dirs";
+    public static final String YARN_TIMELINE_LEVELDB_STORE = "yarn.timeline-service.leveldb-timeline-store.path";
+
     public static final String HDFS_SITE = "hdfs-site";
-    public static final String YARN_NODEMANAGER_LOCAL_DIRS = "yarn.nodemanager.local-dirs";
-    public static final String YARN_NODEMANAGER_LOG_DIRS = "yarn.nodemanager.log-dirs";
-    public static final String HDFS_DATANODE_DATA_DIRS = "dfs.datanode.data.dir";
+    public static final String HDFS_DN_DATA_DIRS = "dfs.datanode.data.dir";
+    public static final String HDFS_NN_CHECKPOINT_DIR = "dfs.namenode.checkpoint.dir";
+    public static final String HDFS_NN_NAME_DIR = "dfs.namenode.name.dir";
+    public static final String HDFS_JN_EDITS_DIR = "dfs.journalnode.edits.dir";
+
+    public static final String HBASE_SITE = "hbase-site";
+    public static final String HBASE_TMP_DIR = "hbase.tmp.dir";
+
+    public static final String STORM_SITE = "storm-site";
+    public static final String STORM_LOCAL_DIR = "storm.local.dir";
+
+    public static final String OOZIE_ENV = "oozie-env";
+    public static final String OOZIE_DATA_DIR = "oozie_data_dir";
+
+    public static final String ZOOKEEPER_CFG = "zoo.cfg";
+    public static final String ZOOKEEPER_DATA_DIR = "dataDir";
+
+    public static final String FALCON_STARTUP = "falcon-startup.properties";
+    public static final String FALCON_GRAPH_PATH = "*.falcon.graph.serialize.path";
+    public static final String FALCON_GRAPH_DIR = "*.falcon.graph.storage.directory";
+
+    private static Map<String, Map<String, List<String>>> requiredGlobalConfigs = new HashMap<>();
+
+    static {
+        requiredGlobalConfigs.put("YARN", singletonMap(YARN_SITE, asList(YARN_TIMELINE_LEVELDB_STORE)));
+        requiredGlobalConfigs.put("HDFS", singletonMap(HDFS_SITE, asList(HDFS_NN_CHECKPOINT_DIR, HDFS_NN_NAME_DIR, HDFS_JN_EDITS_DIR)));
+        requiredGlobalConfigs.put("FALCON", singletonMap(FALCON_STARTUP, asList(FALCON_GRAPH_PATH, FALCON_GRAPH_DIR)));
+        requiredGlobalConfigs.put("HBASE", singletonMap(HBASE_SITE, asList(HBASE_TMP_DIR)));
+        requiredGlobalConfigs.put("OOZIE", singletonMap(OOZIE_ENV, asList(OOZIE_DATA_DIR)));
+        requiredGlobalConfigs.put("STORM", singletonMap(STORM_SITE, asList(STORM_LOCAL_DIR)));
+        requiredGlobalConfigs.put("ZOOKEEPER", singletonMap(ZOOKEEPER_CFG, asList(ZOOKEEPER_DATA_DIR)));
+    }
 
     @Inject
     private HostGroupRepository hostGroupRepository;
 
-    public Map<String, Map<String, Map<String, String>>> getConfiguration(Stack stack) {
+    public Map<String, Map<String, String>> getGlobalConfiguration(Stack stack) throws IOException {
+        Map<String, Map<String, String>> config = new HashMap<>();
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
ObjectMapper is thread safe after configuration. so basically we can create a spring singleton and just inject it here.